ecdsa/src/ecdsa.rs

```diff
@@ -336,24 +353,60 @@ mod tests {
                 assert_eq!(r, r_candidate);
             }
 
+            (public_key, r, s, msg_hash)
+        }
+
+        fn generate_invalid_inputs<C: CurveAffine, N: FromUniformBytes<64> + Ord>() -> (C, C::Scalar, C::Scalar, C::Scalar) {
+            let (public_key, _, s, msg_hash) = generate_valid_inputs::<C, N>();
+            // Set the value of r incorrectly
+            (public_key, s, s, msg_hash)
```

Comment:
Some additional test cases would be great. Also, because the signature data can contain any bytes, it would be great to test if the signature points are outside the field. What do you think the current behavior for that would be?

Reply:
If the coordinates of the signature points are outside the field, it indicates that the signature is not compliant with the requirements of ECDSA over the secp256k1 curve. In such a case, the signature would be considered invalid, and the transaction would be rejected by the network when attempting to verify the signature.

Reply:
We still have to support it though, any data can be made part of our transaction data! Also `ecrecover` needs the check. It seems like `assign_integer` will do a range check to make sure it's inside the field, so looks like we'll have to do a soft `<` check before that happens. If the points are not inside the field the signature should also be marked as invalid.

If we could follow the ecrecover implementation that'd be great. From the Ethereum yellow paper:
> 1. if v /∈ {27, 28} ∨ r = 0 ∨ r ≥ secp256k1n ∨ s = 0 ∨ s ≥ secp256k1n
> 2. if ECDSARECOVER(h, v − 27, r, s) = ∅

We could also do some of the 1. checks elsewhere if you think it's easier.

Reply:
`r`, `s` and `v` soft range checks still TODO?